shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala

```diff
@@ -137,13 +137,23 @@ class Spark310Shims extends Spark301Shims {
           override def tagPlanForGpu(): Unit = GpuFileSourceScanExec.tagSupport(this)
 
           override def convertToGpu(): GpuExec = {
+            val sparkSession = wrapped.relation.sparkSession
+            val options = wrapped.relation.options
             val newRelation = HadoopFsRelation(
               wrapped.relation.location,
               wrapped.relation.partitionSchema,
               wrapped.relation.dataSchema,
               wrapped.relation.bucketSpec,
               GpuFileSourceScanExec.convertFileFormat(wrapped.relation.fileFormat),
-              wrapped.relation.options)(wrapped.relation.sparkSession)
+              options)(sparkSession)
+            val isParquet = newRelation.fileFormat match {
+              case _: ParquetFileFormat => true
+              case _: GpuReadParquetFileFormat => true
+              case _ => false
+            }
+            val canUseSmallFileOpt = (isParquet && conf.isParquetSmallFilesEnabled &&
+              !(options.getOrElse("mergeSchema", "false").toBoolean ||
```

Comment:
Just FYI schema evolution can happen even without `mergeSchema`.  If the user just passes in their own schema you can hit the same thing.

Reply:
thanks, filed followon issue https://github.com/NVIDIA/spark-rapids/issues/608